ss/pebbledb/iterator.go

```diff
@@ -115,23 +115,22 @@ func (itr *iterator) Value() []byte {
 	return slices.Clone(val)
 }
 
-func (itr *iterator) Next() {
+func (itr *iterator) NextForward() {
+	if !itr.source.Valid() {
+		itr.valid = false
+		return
+	}
+
 	currKey, _, ok := SplitMVCCKey(itr.source.Key())
 	if !ok {
 		// XXX: This should not happen as that would indicate we have a malformed
 		// MVCC key.
 		panic(fmt.Sprintf("invalid PebbleDB MVCC key: %s", itr.source.Key()))
 	}
 
-	var next bool
-	if itr.reverse {
-		// Since PebbleDB has no PrevPrefix API, we must manually seek to the next
-		// key that is lexicographically less than the current key.
-		next = itr.source.SeekLT(MVCCEncode(currKey, 0))
-	} else {
-		// move the cursor to the next key
-		next = itr.source.NextPrefix()
-	}
+	fmt.Printf("currKey %s\n", string(currKey))
```

Comment:
There's still debug lines in this PR

Reply:
Yea, as mentioned I'm leaving these in for now. I'll change this to draft and re-open when more tests / sanity checks are complete and this is all cleaned up.